src/libs/ajax/loaded-data/useCachedData.ts

```diff
@@ -0,0 +1,31 @@
+import { Atom, LoadedState, ReadyState } from '@terra-ui-packages/core-utils';
+import { useLoadedDataEvents } from 'src/libs/ajax/loaded-data/useLoadedData';
+import { useSettableStore } from 'src/libs/react-utils';
+
+export const useCachedData = <S, HookResult extends LoadedState<S, unknown>>(
```

Comment:
Not keen on this name since, to me, "cache" implies that it's returning the value from the store instead of making the network request. But the state of the store has no effect on the network call. Maybe something along the lines of `useSyncedStore`?

Reply:
Eh, I guess I can kind of see it. But it depends a lot on the consumer. When used with `useAutoLoadedData`, it serves as a stale while revalidate cache.

When used with `useLoadedData`, its purpose is more limited to syncing the store. A consumer could use `useLoadedData` and `useStore` directly to have the cache functionality.

This one is a little weird anyway, since a hook wrapping a hook just looks off. I'm somewhat tempted to call this machinery for creating a hook rather than a hook itself.

Reply:
fair enough, and I've had my brain in the same space as well... suggesting useFocusedWorkspaces or something like that to indicate it's the workspaces the app is focusing on.... something to indicate it is more then just a data call wrapper... but as you said, more named towards the consumer's usage.

Reply:
hmmm... yeah.  Tho I am trying to avoid higher-order-functions, etc... since those seem to add cognitive load....

Reply:
Don't all custom hooks wrap hooks, by definition?

Reply:
Yes. I should have clarified, I was referring to in the call.

This pattern is unusual:
```ts
useSomething(() => useSomethingElse())
```

Reply:
switched useCachedData to withCachedData wrapper hook that keeps inner-hook signature for the consumer.  See latest commit.